### ISSUE TITLE
GLES2:  Shader compilation errors on Android

### ISSUE THREAD

**Issue by kosz78** [godotengine/godot#21109]
**Godot version:**
Godot 3.1 6936cbc24e7d675c685b43bd86fabc9d572fa178

**OS/device including version:**
Android 6.0, Mali-T720

**Issue description:**
In simple godot project i have a lot of errors on shader compilation under Android for GLES2.
```
**ERROR**: CanvasShaderGLES2: Fragment shader compilation failed:
6:6: L0002: Undeclared variable 'f'
6:7: L0002: Undeclared variable 'm_smoothed_alpha'
	
   At: drivers\gles2\shader_gles2.cpp:457:get_current_version() - CanvasShaderGLES2: Fragment shader compilation failed:
6:6: L0002: Undeclared variable 'f'
6:7: L0002: Undeclared variable 'm_smoothed_alpha'
	
**ERROR**: Method/Function Failed, returning: __null
   At: drivers\gles2\shader_gles2.cpp:466:get_current_version() - Method/Function Failed, returning: __null
**ERROR**: Condition ' !version ' is true. returned: false
   At: drivers\gles2\shader_gles2.cpp:133:bind() - Condition ' !version ' is true. returned: false
   At: drivers\gles2/shader_gles2.h:509:_get_uniform() - Condition ' !version ' is true. returned: -1
**ERROR**: SceneShaderGLES2: Fragment shader compilation failed:
1:1: P0003: Extension 'GL_ARB_shader_texture_lod' not supported
1:54: L0002: No matching function for call to 'texture2DLod'

   At: drivers\gles2\shader_gles2.cpp:457:get_current_version() - SceneShaderGLES2: Fragment shader compilation failed:
1:1: P0003: Extension 'GL_ARB_shader_texture_lod' not supported
1:54: L0002: No matching function for call to 'texture2DLod'

**ERROR**: Method/Function Failed, returning: __null
   At: drivers\gles2\shader_gles2.cpp:466:get_current_version() - Method/Function Failed, returning: __null
**ERROR**: Condition ' !version ' is true. returned: false
   At: drivers\gles2\shader_gles2.cpp:133:bind() - Condition ' !version ' is true. returned: false
**ERROR**: SceneShaderGLES2: Fragment shader compilation failed:
1:1: P0003: Extension 'GL_ARB_shader_texture_lod' not supported
1:54: L0002: No matching function for call to 'texture2DLod'

   At: drivers\gles2\shader_gles2.cpp:457:get_current_version() - SceneShaderGLES2: Fragment shader compilation failed:
1:1: P0003: Extension 'GL_ARB_shader_texture_lod' not supported
1:54: L0002: No matching function for call to 'texture2DLod'

**ERROR**: Method/Function Failed, returning: __null
   At: drivers\gles2\shader_gles2.cpp:466:get_current_version() - Method/Function Failed, returning: __null
**ERROR**: Condition ' !version ' is true. returned: false
   At: drivers\gles2\shader_gles2.cpp:133:bind() - Condition ' !version ' is true. returned: false
```

**Minimal reproduction project:**
[godot-gles2-test.zip](https://github.com/godotengine/godot/files/2296726/godot-gles2-test.zip)

**Comment by akien-mga** [comment]
CC @karroffel

**Comment by dertom95** [comment]
Same here (moto g2/adreno 305)

Full logcat: https://pastebin.com/UAZ75QN8

**Comment by TheOnlyJoey** [comment]
This is a known issue, Keep in mind Adreno is tricky on GLES3 due to #12816

**Comment by dertom95** [comment]
Noone is talking about gles3. All this is about gles2!

**Comment by akien-mga** [comment]
Assuming fixed by #21318 - if you still reproduce the exact same issue, please comment here :)
